pkg/core/system.go

```diff
@@ -55,58 +63,184 @@ var (
+			return false, nil
 		}
-		fmt.Print("Istio components installed\n\n")
-	} else {
-		if !options.Force {
-			answer, err := confirm("Istio is already installed, do you want to install the Knative components for riff?")
-			if err != nil {
-				return false, err
+		if old != nil {
+			return true, errors.New(fmt.Sprintf("%s already installed", env.Cli.Name))
+		}
+		_, err = c.crdClient.Create(manifest)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err == wait.ErrWaitTimeout {
+		return nil, errors.New(fmt.Sprintf("timed out creating %s custom resource defiition", env.Cli.Name))
+	}
+	return manifest, err
+}
+
+func getElementContaining(array []string, substring string) string {
+	for _, s := range array {
+		if strings.Contains(s, substring) {
+			return s
+		}
+	}
+	return ""
+}
+
+func (c *client) installAndCheckResources(manifest *crd.Manifest, options SystemInstallOptions) error {
+	for _,resource := range manifest.Spec.Resources.System {
+		err := c.installResource(resource, options)
+		if err != nil {
+			return err
+		}
+		err = c.checkResource(resource)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *client) installResource(res crd.Resource, options SystemInstallOptions) error {
+	if res.Path == "" {
+		return errors.New("cannot install anything other than a url yet")
+	}
+	fmt.Printf("installing %s from %s...", res.Name, res.Path)
+	yaml, err := fileutils.Read(res.Path, filepath.Dir(options.Manifest))
+	if err != nil {
+		return err
+	}
+	if options.NodePort {
+		yaml = bytes.Replace(yaml, []byte("type: LoadBalancer"), []byte("type: NodePort"), -1)
+	}
+	// TODO HACK: use the RESTClient to do this
+	kubectl := kubectl.RealKubeCtl()
+	istioLog, err := kubectl.ExecStdin([]string{"apply", "-f", "-"}, &yaml)
+	if err != nil {
+		fmt.Printf("%s\n", istioLog)
+		if strings.Contains(istioLog, "forbidden") {
+			fmt.Print(`It looks like you don't have cluster-admin permissions.
+
+To fix this you need to:
+ 1. Delete the current failed installation using:
+      ` + env.Cli.Name + ` system uninstall --istio --force
+ 2. Give the user account used for installation cluster-admin permissions, you can use the following command:
+      kubectl create clusterrolebinding cluster-admin-binding \
+        --clusterrole=cluster-admin \
+        --user=<install-user>
+ 3. Re-install ` + env.Cli.Name + `
+
+`)
+		}
+		return err
+	}
+	return nil
+}
+
+// TODO this only supports checking Pods for phases
+func (c *client) checkResource(resource crd.Resource) error {
+	cnt := 1
+	for _, check := range resource.Checks {
+		var ready bool
+		var err error
+		for i := 0; i< 360; i++ {
+			if strings.EqualFold(check.Kind, "Pod") {
+				ready, err = c.isPodReady(check, resource.Namespace)
+				if err != nil {
+					return err
+				}
+				if ready {
+					break
+				}
+			} else {
+				return errors.New("only Kind:Pod supported for resource checks")
```

Comment:
How do we check that a LoadBalancer Service is ready?

The world is more than Pods.